app/views/admin/attachments/_form.html.erb

```diff
@@ -43,20 +41,22 @@
         } %>
       </div>
 
-      <%= render "components/govspeak-editor", {
-        label: {
-          heading_size: "l",
-          text: "Body (required)"
-        },
-        name: "attachment[govspeak_content_attributes][body]",
-        rows: 20,
-        id: "attachment_govspeak_content_body",
-        value: form.object.govspeak_content.body,
-        error_items: errors_for(attachment.errors, :"govspeak_content.body"),
-      } %>
+      <div class="govuk-!-margin-bottom-8 attachment-form__body" dir="<%= form.object.rtl_locale? ? "rtl" : "ltr" %>">
```

Comment:
I have a slight concern that we're setting `dir="rtl"` too 'high up' in the DOM.

This `<div class="attachment-form__body">` doesn't _only_ contain RTL user input (i.e. the `<textarea>` and rendered preview) – it also contains UI elements such as the `<label for="attachment_govspeak_content_body">` and the `<button>Preview</button>`.

I don't think those UI elements should switch to right-to-left just because the user input does. It seems semantically 'wrong' to say that the UI is switching direction.

For example, this is how the UI changes when I toggle that `dir` attribute.

| `dir="ltr"` | `dir="rtl"` |
| --- | --- |
| ![Left to right](https://user-images.githubusercontent.com/7735945/201704686-06b0e404-e4db-4961-a18a-6d949b2c8cb5.png) | ![Right to left](https://user-images.githubusercontent.com/7735945/201704756-81075355-024b-4a40-bf1b-c65b945aa5ed.png) |

**Disclaimer:** I've done this in my browser's dev tools in the integration environment, so I'm not actually running your branch. But I think this is how it'll display nonetheless.

Reply:
@ollietreend Yes, fair point, this is a hangover from making changes "like-for-like". Given the update in the means by which the text direction is handled it's probably a sensible enhancement.